front/src/applications/stdcm/utils/fomatConflicts.ts

```diff
@@ -0,0 +1,82 @@
+import dayjs from 'dayjs';
+
+import type { ManageTrainSchedulePathProperties } from 'applications/operationalStudies/types';
+import type { Conflict } from 'common/api/osrdEditoastApi';
+import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+
+interface ConflictWithInterval {
+  trainIds?: number[];
+  startTime: string;
+  endTime: string;
+  endDate: string;
+  startDate: string;
+  conflictType: string;
+  waypointBefore: string | null;
+  waypointAfter: string | null;
+}
+
+// Helper function to find the closest operational point before or after a given position
+const findClosestOperationalPoint = (
+  operationalPoints: SuggestedOP[],
+  position: number,
+  direction: 'before' | 'after'
+) => {
+  if (direction === 'before') {
+    const pointsBefore = operationalPoints?.filter((point) => point?.positionOnPath <= position);
+    return pointsBefore
+      ? pointsBefore?.reduce((prev, curr) =>
+          curr.positionOnPath > prev.positionOnPath ? curr : prev
+        )
+      : null;
+  }
+  const pointsAfter = operationalPoints?.filter((point) => point.positionOnPath >= position);
+  return pointsAfter
+    ? pointsAfter?.reduce((prev, curr) => (curr.positionOnPath < prev.positionOnPath ? curr : prev))
+    : null;
```

Comment:
I think we could do something simpler

```suggestion
  if (direction === 'before') {
    const firstOpIndexInConflict = operationalPoints.findIndex((op) => op.positionOnPath >= position); // We know we will find an op (at worse it will be op at position 0)
    return operationalPoints![firstOpIndexInConflict - 1] // return the last op before the beginning of the conflict zone
  }
  return operationalPoints.find((op) => op.positionOnPath >= position); // return the first op after the end of the conflict zone (at worse it will be the last op from the path)
```

Reply:
if it's "at worse it will be op at position 0" wouldn't it result a bug here `operationalPoints![firstOpIndexInConflict - 1]` ?

Reply:
Yes you're right, we need to handle this case : 

```
if (direction === 'before') {
    const firstOpIndexInConflict = operationalPoints.findIndex(
      (op) => op.positionOnPath >= position
    ); // We know we will find an op (at worse it will be op at position 0)
    return operationalPoints[firstOpIndexInConflict - 1] ?? operationalPoints.at(0)!; // return the last op before the beginning of the conflict zone
  }
  return operationalPoints.find((op) => op.positionOnPath >= position)!; // return the first op after the end of the conflict zone (at worse it will be the last op from the path)
```